core/services/relay/evm/mercury/persistence_manager.go

```diff
@@ -0,0 +1,130 @@
+package mercury
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/mercury/wsrpc/pb"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+var (
+	flushDeletesFrequency = time.Second
+	pruneFrequency        = time.Hour
+)
+
+type PersistenceManager struct {
+	lggr logger.Logger
+	orm  *ORM
+
+	once   utils.StartStopOnce
+	stopCh utils.StopChan
+	wg     sync.WaitGroup
+
+	deleteMu    sync.Mutex
+	deleteQueue []*pb.TransmitRequest
+}
+
+func NewPersistenceManager(lggr logger.Logger, orm *ORM) *PersistenceManager {
+	return &PersistenceManager{
+		lggr:   lggr.Named("MercuryPersistenceManager"),
+		orm:    orm,
+		stopCh: make(chan struct{}),
+	}
+}
+
+func (pm *PersistenceManager) Start(ctx context.Context) error {
+	return pm.once.StartOnce("MercuryPersistenceManager", func() error {
+		pm.wg.Add(2)
+		go pm.runFlushDeletesLoop()
+		go pm.runPruneLoop()
+		return nil
+	})
+}
+
+func (pm *PersistenceManager) Close() error {
+	return pm.once.StopOnce("MercuryPersistenceManager", func() error {
+		close(pm.stopCh)
+		pm.wg.Wait()
+		return nil
+	})
+}
+
+func (pm *PersistenceManager) Insert(ctx context.Context, req *pb.TransmitRequest, reportCtx ocrtypes.ReportContext) error {
+	return pm.orm.InsertTransmitRequest(req, reportCtx, pg.WithParentCtx(ctx))
+}
+
+func (pm *PersistenceManager) Delete(ctx context.Context, req *pb.TransmitRequest) error {
+	return pm.orm.DeleteTransmitRequests([]*pb.TransmitRequest{req}, pg.WithParentCtx(ctx))
+}
+
+func (pm *PersistenceManager) AsyncDelete(req *pb.TransmitRequest) {
+	pm.addToDeleteQueue(req)
+}
+
+func (pm *PersistenceManager) Load(ctx context.Context) ([]*Transmission, error) {
+	return pm.orm.GetTransmitRequests(pg.WithParentCtx(ctx))
+}
+
+func (pm *PersistenceManager) runFlushDeletesLoop() {
+	defer pm.wg.Done()
+
+	ctx, cancel := pm.stopCh.Ctx(context.Background())
+	defer cancel()
+
+	ticker := time.NewTicker(flushDeletesFrequency)
+	for {
+		select {
+		case <-ctx.Done():
+			ticker.Stop()
+			return
+		case <-ticker.C:
+			pm.lggr.Trace("Deleting queued requests from transmit requests table")
+			queuedReqs := pm.resetDeleteQueue()
+			if err := pm.orm.DeleteTransmitRequests(queuedReqs, pg.WithParentCtx(ctx)); err != nil {
```

Comment:
You might consider `pg.WithLongQueryTimeout` here

Reply:
At the current flush frequency of 1s, we'd expect ~100 reports to be deleted per loop. I'd expect 100 indexed deletes to take ~100ms so the default timeout of 10s should be appropriate.